main.js

```diff
@@ -161,6 +160,14 @@ onReady = () => {
 
     // Enable the Swarm protocol
     protocol.registerHttpProtocol('bzz', (request, callback) => {
+        if (!store.getState().settings.swarmEnabled) {
+            let error = global.i18n.t('mist.errors.swarm.notEnabled');
+            dialog.showErrorBox('Error', error);
+            callback({ error: error });
+            store.dispatch({ type: '[MAIN]:PROTOCOL:ERROR', payload: { protocol: 'bzz', error: error } });
```

Comment:
nit: with ES6, you can reduce identical object key/value pairs to just the key.
`callback({ error });`
`... payload: { protocol: 'bzz', error } ...`

Reply:
nice, that's awesome. thanks for the tip